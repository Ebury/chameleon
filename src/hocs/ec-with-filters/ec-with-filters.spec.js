import { mount, createLocalVue, createWrapper } from '@vue/test-utils';
import withFilters from './ec-with-filters';

describe('EcWithFilters', () => {
  function mountEcWithFilters(filters, props, mountOpts) {
    const localVue = createLocalVue();

    const Component = localVue.extend({
      props: ['filters'],
      render() {
        return (<div>
          <span># of filters: { this.filters.length }</span>
          {this.filters.map(filter => <span>{filter.name}</span>)}
        </div>);
      },
    });

    const hocWrapper = mount(withFilters(Component, filters), {
      localVue,
      propsData: { ...props },
      ...mountOpts,
    });

    const componentWrapper = createWrapper(hocWrapper.vm.$children[0].$vnode);

    return { hocWrapper, componentWrapper };
  }

  it('should render properly', () => {
    const { hocWrapper } = mountEcWithFilters();
    expect(hocWrapper.element).toMatchSnapshot();
  });

  it('should bound given filters', async () => {
    const filters = [{
      name: 'test1',
    }, {
      name: 'test2',
    }];

    const { hocWrapper } = mountEcWithFilters(filters);
    expect(hocWrapper.element).toMatchSnapshot();
  });
});
