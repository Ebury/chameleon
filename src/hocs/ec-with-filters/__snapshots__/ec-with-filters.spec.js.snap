// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcWithFilters should bound given filters 1`] = `
<div>
  <span>
    # of filters: 2
  </span>
  <span>
    test1
  </span>
  <span>
    test2
  </span>
</div>
`;

exports[`EcWithFilters should render properly 1`] = `
<div>
  <span>
    # of filters: 0
  </span>
</div>
`;
